File: pdu/python/tf2_msgs/pdu_cdr_conv_TFMessage.py
from .pdu_pytype_TFMessage import TFMessage
from ..pdu_cdr_runtime import CdrReader, CdrWriter

from ..std_msgs.pdu_cdr_conv_Header import *
from ..geometry_msgs.pdu_cdr_conv_Quaternion import *
from ..builtin_interfaces.pdu_cdr_conv_Time import *
from ..geometry_msgs.pdu_cdr_conv_Transform import *
from ..geometry_msgs.pdu_cdr_conv_TransformStamped import *
from ..geometry_msgs.pdu_cdr_conv_Vector3 import *



def py_to_cdr_body_TFMessage(writer: CdrWriter, src: TFMessage):
    writer.write_sequence_length(src.transforms)
    for elem in src.transforms:
        py_to_cdr_body_TransformStamped(writer, elem)


def cdr_body_to_py_TFMessage(reader: CdrReader, dst: TFMessage):
    dst.transforms = []
    for _ in range(reader.read_uint32()):
        elem = TransformStamped()
        cdr_body_to_py_TransformStamped(reader, elem)
        dst.transforms.append(elem)
    return dst


def py_to_cdr_TFMessage(src: TFMessage) -> bytes:
    writer = CdrWriter()
    writer.write_encapsulation()
    py_to_cdr_body_TFMessage(writer, src)
    return writer.bytes()


def cdr_to_py_TFMessage(cdr_payload) -> TFMessage:
    reader = CdrReader(cdr_payload)
    reader.read_encapsulation()
    dst = TFMessage()
    return cdr_body_to_py_TFMessage(reader, dst)

File: pdu/javascript/tf2_msgs/pdu_cdr_conv_TFMessage.js
import { TFMessage } from './pdu_jstype_TFMessage.js';
import { PduCdrWriter, PduCdrReader } from '../pdu_cdr_runtime.js';
import { Header } from '../std_msgs/pdu_jstype_Header.js';
import { PduHeaderConverter } from '../std_msgs/pdu_cdr_conv_Header.js';
import { Quaternion } from '../geometry_msgs/pdu_jstype_Quaternion.js';
import { PduQuaternionConverter } from '../geometry_msgs/pdu_cdr_conv_Quaternion.js';
import { Time } from '../builtin_interfaces/pdu_jstype_Time.js';
import { PduTimeConverter } from '../builtin_interfaces/pdu_cdr_conv_Time.js';
import { Transform } from '../geometry_msgs/pdu_jstype_Transform.js';
import { PduTransformConverter } from '../geometry_msgs/pdu_cdr_conv_Transform.js';
import { TransformStamped } from '../geometry_msgs/pdu_jstype_TransformStamped.js';
import { PduTransformStampedConverter } from '../geometry_msgs/pdu_cdr_conv_TransformStamped.js';
import { Vector3 } from '../geometry_msgs/pdu_jstype_Vector3.js';
import { PduVector3Converter } from '../geometry_msgs/pdu_cdr_conv_Vector3.js';


export class PduTFMessageConverter {
    /**
     * @param {PduCdrWriter} writer
     * @param { TFMessage } src
     */
    static to_cdr_body(writer, src) {
        writer.write_sequence_length(src.transforms);
        for (const elem of src.transforms) {
            PduTransformStampedConverter.to_cdr_body(writer, elem);
        }
    }

    /**
     * @param {PduCdrReader} reader
     * @param { TFMessage } dst
     * @returns { TFMessage }
     */
    static cdr_body_to_js(reader, dst) {
        dst.transforms = [];
        for (let i = 0, len = reader.read_uint32(); i < len; i++) {
            dst.transforms.push(PduTransformStampedConverter.cdr_body_to_js(reader, new TransformStamped()));
        }
        return dst;
    }

    /**
     * @param { TFMessage } src
     * @returns {ArrayBuffer}
     */
    static to_cdr(src) {
        const writer = new PduCdrWriter();
        writer.write_encapsulation();
        this.to_cdr_body(writer, src);
        return writer.get_buf();
    }

    /**
     * @param {ArrayBuffer|ArrayBufferView} cdrPayload
     * @returns { TFMessage }
     */
    static from_cdr(cdrPayload) {
        const reader = new PduCdrReader(cdrPayload);
        reader.read_encapsulation();
        return this.cdr_body_to_js(reader, new TFMessage());
    }
}
